importer-declarative.md

```diff
@@ -0,0 +1,164 @@
+By leveraging selector expressions it is possible to create complex transformations without writing any code.
+
+#### Block config expressions
+
+The `buildBlockConfig` function of the `Transformer` class can be used to create a block configuration object by mapping from an object of CSS selector expressions.
+
+An expression item could be a single name/value pair that maps directly. An item could be a simple expression where the value is obtained from a single CSs selector. 
+Finally, multiple expressions can be defined in an array to allow values to be conditionally applied based on the page structure.
+
+Example:
+```json
+{
+    "simple_property": "Static value",
+    "single_expression": "[name=\"keywords\"]",
+    "multiple_expressions": [
+        [":scope:has(.is-blog)", "Webinars"],
+        [":scope:has(.is-blog)", ".blog-category"]
+    ]
+}
+```
+
+#### Block cell expressions
+
+Similarly, the `buildBlockCells` function is able to map a two-dimensional array of CSS selector expressions to an array of elements that should be added to a block.
+
+Example: Creates a block with one row and two columns. The first column contains a div that includes an image. The second column contains the last child div under the source element.
+```json
+[
+  ["div:has(p > img)", ":scope > div:last-child"]
+]
+```
+
+### Built-in block parsers
+
+#### Default block parser
+
+The default block parser is used when no custom block parser is defined. It is able to create many different blocks, however, by consuming complex selector expressions.
+The selector expressions used by the default block parser must be defined as a `cells` property in the `params` object.
+
+eg.
+```json
+{
+  "type": "my_block",
+  "selectors": [
+    ".selector div:has(div > p > img)",
+    ".selector > div > div:first-of-type:has(div > img)"
+  ],
+  "params": {
+    "cells": [
+      [[".col1_selector_1", ".col1_selector_2"], [".col2_selector"]]
+    ]
+  }
+}
+```
+
+#### Metadata block parser
+
+Metadata is a built-in block concept and has its own parser that is able to extract metadata from the source DOM.
+Since the Metadata block uses the block config format, a block config expression can be included with metadata parsing. 
+The `metadata` parser will use selector expressions included with the `metadata` property of the `params` object.  
+Title, Description, and Image metadata will automatically be included in the block and do not need to be defined in the block config expression.
+
+eg.
+```json
+{
+  "type": "metadata",
+  "target": "append",
+  "params": {
+    "metadata": {
+      "keywords": "[name=\"keywords\"]",
+      "Publication Date": "[property=\"og:article:published_time\"]",
+      "category": [
+        [
+          ":scope:has(.is-blog .post-list) .webinar-speaker-img",
+          "Webinars"
+        ],
+        [
+          ":scope:has(.is-blog .post-list)",
+          ""
+        ]
```

Comment:
I like this IFTT approach here.  Are they done in order and if there is a match the rest is ignored?
The example (empty string) confuses me a little.

Reply:
Yes, first match is used. Any others are ignored. Sometimes you may want an entry added to a block for an author to fill out later, the empty string allows such a scenario.